editreport_form.php

```diff
@@ -102,8 +130,14 @@ public function definition() {
             $mform->setType('courseid', PARAM_INT);
         }
 
+        // Submit button string.
+        $submitstring = get_string('add');
+        if (isset($this->_customdata['report']->id) && $this->_customdata['report']->id) {
```

Comment:
This should collapse down to empty(). See https://www.php.net/manual/en/function.empty.php